### PR TITLE
Remove pre go1.8 code and cleanup

### PR DESCRIPTION
**Pull request by harshavardhana**
We don't need certain go1.7.x custom code anymore, since
we have migrated to go1.8

## Description
Remove pre go1.8 code and cleanup

## Motivation and Context
Migrate to go1.8 cleanup.

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/notify-webhook.go [hunk 1]
```diff
@@ -122,50 +122,20 @@ func lookupEndpoint(urlStr string) error {
...
+		return err
 	}
 
-	// Succes.
-	return err
+	// HTTP status OK/NoContent.
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
+		return fmt.Errorf("Unable to lookup webhook endpoint %s response(%s)", urlStr, resp.Status)
+	}
+	return resp.Body.Close()
```
Comment (vadmeste) [inline cmd/notify-webhook.go]: Shouldn't we also close resp.Body when status code != 2xx ?

Comment (harshavardhana) [reply cmd/notify-webhook.go]: We could potentially do that..

**Comment by harshavardhana** [comment]
//cc @vadmeste @deekoder looking for reviews..

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by balamurugana — APPROVED** [review]
LGTM

**Pull request merged**